closed/src/java.base/share/classes/openj9/internal/security/RestrictedSecurity.java

```diff
@@ -1070,23 +924,604 @@ boolean isRestrictedProviderAllowed(String providerName) {
+                parseProvider(providerInfo, i, false, isFIPS140_2);
+            }
+        }
+
+        /**
+         * Load restricted security properties.
+         */
+        private void loadProperties(String profileID, List<String> allInfo) {
+            if (debug != null) {
+                debug.println("\tLoading properties of restricted security profile.");
+            }
+
+            setProperty("descName", profileID + ".desc.name", allInfo);
+            if (setProperty("descIsDefaultString", profileID + ".desc.default", allInfo)) {
+                descIsDefault = Boolean.parseBoolean(profileProperties.get("descIsDefaultString"));
+            }
+            if (setProperty("descIsFIPSString", profileID + ".desc.fips", allInfo)) {
+                descIsFIPS = Boolean.parseBoolean(profileProperties.get("descIsFIPSString"));
+            }
+            setProperty("descNumber", profileID + ".desc.number", allInfo);
+            setProperty("descPolicy", profileID + ".desc.policy", allInfo);
+            setProperty("descSunsetDate", profileID + ".desc.sunsetDate", allInfo);
+
+            setProperty("jdkTlsDisabledNamedCurves",
+                    profileID + ".tls.disabledNamedCurves", allInfo);
+            setProperty("jdkTlsDisabledAlgorithms",
+                    profileID + ".tls.disabledAlgorithms", allInfo);
+            setProperty("jdkTlsEphemeralDHKeySize",
+                    profileID + ".tls.ephemeralDHKeySize", allInfo);
+            setProperty("jdkTlsLegacyAlgorithms",
+                    profileID + ".tls.legacyAlgorithms", allInfo);
+            setProperty("jdkCertpathDisabledAlgorithms",
+                    profileID + ".jce.certpath.disabledAlgorithms", allInfo);
+            setProperty("jdkSecurityLegacyAlgorithms",
+                    profileID + ".jce.legacyAlgorithms", allInfo);
+            setProperty("keyStoreType",
+                    profileID + ".keystore.type", allInfo);
+            setProperty("keyStore",
+                    profileID + ".javax.net.ssl.keyStore", allInfo);
+
+            setProperty("jdkSecureRandomProvider",
+                    profileID + ".securerandom.provider", allInfo);
+            setProperty("jdkSecureRandomAlgorithm",
+                    profileID + ".securerandom.algorithm", allInfo);
+            setProperty("jdkFipsMode",
+                    profileID + ".fips.mode", allInfo);
+
+            if (debug != null) {
+                debug.println("\tProperties of restricted security profile successfully loaded.");
+            }
+        }
+
+        private void setConstraints(String providerName, String providerInfo, boolean providerChanged) {
+            if (debug != null) {
+                debug.println("\t\tLoading constraints for security provider: " + providerName);
+            }
+
+            List<Constraint> constraints = new ArrayList<>();
+
+            providerInfo = providerInfo.replaceAll("\\s+", "");
```

Comment:
We haven't provided an example `java.security` that has significant whitespace within constraints, but how do we know that we will never need them? For example, suppose a constraint needs to specify an attribute value which must include a space?

Reply:
All of the attributes provided in existing providers are `ImplementedIn=Software`, or single values or lists of things like `ValidationAlgorithm`, `ThreadSafe`, `SupportedKeyClasses`, `KeySize` etc. I haven't found anything else and I cannot think of anything that would need a space. Anything that comes to your mind?

Reply:
I don't have a specific example to offer, but I was thinking along the lines of a "transformation" (the first argument for `Cipher.getInstance(String, String)`) that might need to contain one or more spaces.

Reply:
I haven't encountered a transformation that requires a space anywhere so far. I think we should remove the spaces to be sure that we avoid typos at this point. We can always make another change in the future if such a scenario comes up, even though I don't think that's a likely one.

Reply:
I don't know enough to comment on the likelihood of such a problem, but I couldn't find anything in this repository that would require such spaces, so that seems a reasonable posture.